libs/libarm-optimized-routines/aor.h

```diff
@@ -0,0 +1,226 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+extern "C" {
+#include <stringlib.h>
+}
+
+#include <cstddef>
+#include <cstring>
+
+#define ALWAYS_INLINE __attribute__((__always_inline__))
+
+ALWAYS_INLINE static inline void * inline_memcpy(void * __restrict dst_, const void * __restrict src_, size_t size)
+{
+#if __aarch64__
+#if __ARM_FEATURE_SVE
+    return __memcpy_aarch64_sve(dst_, src_, size);
+#elif __ARM_NEON
+    return __memcpy_aarch64_simd(dst_, src_, size);
+#endif
+    return __memcpy_aarch64(dst_, src_, size);
+#elif __arm__
+    return __memcpy_arm(dst_, src_, size);
+#else
+    return memcpy(dst_, src_, size);
+#endif
+}
```

Comment:
Can this file get the correct macro for `__ARM_FEATURE_SVE` / `__ARM_NEON`, etc?

Reply:
I'm not sure where these macro is defined. Are they defined in the arm system header?

Reply:
Yes, I have verified.

![image](https://github.com/pingcap/tiflash/assets/60744015/a4093942-1dd2-40cc-9a21-115f720fcaeb)


They are introduced by flag, like "+simd" will add macro `__ARM_NEON`.

https://developer.arm.com/documentation/dui0774/l/Other-Compiler-specific-Features/Predefined-macros